pkg/testhelper/accessory.go

```diff
@@ -120,8 +120,21 @@ type Accessory struct {
 	clientFlags PortFlags
 }
 
-// Run begins the accessory process. This call is not blocking.
-func (a *Accessory) Run(t *T, parentCtx context.Context) {
+// run begins the accessory process. Only test/e2e/framework.Run
+// is allowed to call this as it required additional synchronization or your
+// tests might pass incorrectly.
+func (a *Accessory) RunFromFrameworkRunner(t *T, parentCtx context.Context) {
+	a.run(parentCtx, t, t.Fatalf)
+}
+
+// run begins the accessory process. this call is not blocking.
+// Because testing.T does not allow to call Fatalf in a distinct
+// goroutine, this will use Errorf instead.
+func (a *Accessory) Run(t *testing.T) {
+	a.run(context.Background(), t, t.Errorf)
```

Comment:
So will this leak Vault processes or whatever?

Reply:
It doesn't look like, I don't have any stray vault processes after running these tests locally